src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -321,6 +333,15 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                         token);
                 }
             }
+            else if (restoreSource == RestoreOperationSource.Explicit)
+            {
+                // When the user consent is not granted, missing packages may not be restored.
+                // So, we just check for them, and report them as warning(s) on the error list window
+                _logger.Do((l, _) =>
+                {
+                    l.ShowError(Resources.PackageRefNotRestoredBecauseOfNoConsent);
```

Comment:
Should it really be an error? It's more likely an information since customer explicitly disable it.